### DISCUSSION TITLE
Does the RSSI limit apply to bluetooth devices

### DISCUSSION THREAD

**Discussion by wally2511** [cyberman54/ESP32-Paxcounter#817] [Q&A]
I tested the paxcounter at work today and had a huge number of bluetooth devices detect, this is because our standing desks are bluetooth enable. I want to limit the proximity of the devices detect so I fiddle with the RSSI limit value. I tried values from -40 to -110 but it didn't seem to change the number of bluetooth devices detect.

Does the RSSI limit parameter apply to bluetooth RF or only to Wifi?

**Comment by cyberman54** [comment]
yes, it does / should. But you need to use a positive value, e.g. 110 for threshold -110 rssi.

**Comment by wally2511** [comment]
Ok. So I sent "01 42" as a remote command to set rssi limit to 50. I saw
the value on the oled screen change to -50 but it had no effect on the
number of Bluetooth devices detected. Any thoughts?

**Reply by cyberman54** [reply]
@FlorianLudwig can you help here? Maybe a bug in the new libpax lib, or in libpax config settings?

**Reply by oliverbrandmueller** [reply]
AFAIK is RSSI Limiting (as well as MAC Filtering) not (yet) implemented in libpax - please open an discussion/issue at the https://github.com/dbSuS/libpax project

**Comment by wally2511** [comment]
Will do, thanks!

**Comment by cyberman54** [comment]
I added the rssi feature to the libpax lib in the fork used for paxcounter. /cc @spmrider 

@wally2511 Could you please re-test, using current paxcounter software in master branch? Please make sure that a local copy of the libpax library is updated, this should happen automatically in platformio, if not, delete the lib from disk by hand before build the paxcounter code.

**Comment by wally2511** [comment]
I downloaded the current software from the master branch (confirming the
new library was included). I did some testing but couldn't get the RSSI
filtering to work for BLE.
I couldn't work out how I was supposed to offically
define ble_rssi_threshold so tried several options but it didn't take
effect.

**Reply by cyberman54** [reply]
Threshold value is set by a remote command, sent as downlink over the LORAWAN.
The command is 0x01 0x<ab> where <ab> is an positive integer value for (negative) rssi threshold. It is applied as filter on Wifi and BLE signals.

Example: Set threshold -90 rssi (or better) -> rcommand 0x01 0x5A

If you want to make the threshold persistent, add command 0x21 to save setting to NVRAM -> e.g. 0x01 0x05a 0x21

**Comment by wally2511** [comment]
I did some testing this weekend and can confirm the RSSI limit is working.
Thank you and well done!
